sdcm/cluster.py

```diff
@@ -3726,23 +3718,34 @@ def node_setup(_node: BaseNode):
             except Exception:  # pylint: disable=broad-except
                 LOGGER.warning("Failure settings shards for node %s in Argus.", _node)
                 LOGGER.debug("Exception details:\n", exc_info=True)
-            _queue.put((_node, exception_details))
-            _queue.task_done()
+            task_queue.put((_node, exception_details))
+            task_queue.task_done()
+
+        @raise_event_on_failure
+        def node_startup(_node: BaseNode, task_queue: queue.Queue):
```

Comment:
`node_startup` is not executed in parallel/thread context - so there's no need for queues and verification loop later

Reply:
I agree that it can be improved even more, just tried to have less code changes meaning `previous coding approach is mostly kept`.

Reply:
Both functions reuse the same checker-func. So both use queues and both are easy to make it be serial and parallel.